### PR TITLE
GLOB-32441 Add support for bundle dependencies

### PR DESCRIPTION
**Pull request by pokey**

### CODE REVIEW

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/app_hooks/evaluate/app.py
Comment (pokey) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/app_hooks/evaluate/app.py:55]: These will automatically be loaded by virtue of the fact that `active_bundle` and `active_evaluation` point to them

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py
Comment (pokey) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py:42]: @salmanmashayekh to answer your [question](https://github.com/globality-corp/microcosm-sagemaker/pull/33/files#r282148567) from the other PR, it is actually not essential that the bundle artifact be in a particular place.  We could have the bundle artifact in a separate directory.  We just use the one that is in the bigger root artifact for convenience

**Review by piercefreeman — APPROVED** [review]
LGTM.  Still some small dialogue going on in https://github.com/globality-corp/microcosm-sagemaker/pull/37 but approving proactively so it's ready once that PR is merged.

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py
Comment (salmanmashayekh) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py]: One suggestion: why don't we define all of the required setup parameters under the `ExampleBundleTestCase`? We can then call it e.g. `ExampleBundleTestSetup`.

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/commands/evaluate/test_evaluate.py
Comment (salmanmashayekh) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/commands/evaluate/test_evaluate.py]: Do we still need both the `gold-output-artifact` and the `input-artifact` fixtures? If each bundle, even the "master" bundle, has their own subdir and `configuration.json` sits in the parent dir, then one `gold-output` fixture will be enough. Right?

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py
Comment (salmanmashayekh) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py]: `microcosm-sagemaker` is the one that passes the `bundle` argument to this function, so the only things we have control over in here are: a) the input to the `predict` method and b) the expected output. With that in mind, how about we define a set of `(input, expected_output)` pairs in here; and abstract away the `bundle.predict`? 

For example:
https://github.com/globality-corp/ml-nlp-serviceoffering-classifier/blob/master/ml_nlp_serviceoffering_classifier/tests/bundles/test_service_offering_bundle.py#L31
https://github.com/globality-corp/ml-nlp-serviceoffering-classifier/blob/master/ml_nlp_serviceoffering_classifier/tests/bundles/test_classifier_baseline_bundle.py#L23

**Review by salmanmashayekh — APPROVED** [review]

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py
Comment (pokey) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py]: I thought it would be clearer to move the common parameters to the base class, and define the rest only in the test case where they are needed.  This way it is clear exactly what each parameter is used for.  Make sense?

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/commands/evaluate/test_evaluate.py
Comment (pokey) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/commands/evaluate/test_evaluate.py]: My thinking was that in some situations the two could differ, but maybe let's go with the simple case for the cookiecutter.  I'll merge them

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py
Comment (pokey) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py]: Nice call.  Just took a swing at that; lmk what you think

Comment (salmanmashayekh) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/tests/bundles/test_example_bundle.py]: Looks great!

**Pull request merged**
